class.jetpack-gutenberg.php

```diff
@@ -1,78 +1,71 @@
 <?php
+function jetpack_register_block( $type, $args = array() ) {
+	Jetpack_Gutenberg::add_block( $type, $args );
+}
 
 /**
  * General Gutenberg editor specific functionality
  */
 class Jetpack_Gutenberg {
-	/**
-	 * Check if Gutenberg editor is available
-	 *
-	 * @since 6.7.0
-	 *
-	 * @return bool
-	 */
-	public static function is_gutenberg_available() {
-		return function_exists( 'register_block_type' );
+
+	public static $blocks = array();
+	public static function add_block( $type, $args ) {
+		self::$blocks[ $type ] = $args;
 	}
 
-	/**
-	 * Load Gutenberg assets
-	 *
-	 * @since 6.7.0
-	 *
-	 * @return void
-	 */
-	public static function enqueue_block_assets() {
-		if ( ! self::should_load_blocks() ) {
+	public static function load_blocks() {
+		if ( ! self::is_gutenberg_available() ) {
 			return;
 		}
 
-		$rtl = is_rtl() ? '.rtl' : '';
+		if ( self::should_load_blocks() ) {
+			return;
+		}
+		foreach ( self::$blocks as $type => $args ) {
+			register_block_type(
+				'jetpack/' . $type,
+				$args
+			);
+		}
+	}
 
-		/**
-		 * Filter to enable serving blocks via CDN
-		 *
-		 * CDN cache is busted once a day or when Jetpack version changes. To customize it:
-		 *   add_filter( 'jetpack_gutenberg_cdn_cache_buster', function( $version ) { return time(); }, 10, 1 );
-		 *
-		 * @since 6.5.0
-		 *
-		 * @param bool false Whether to load Gutenberg blocks from CDN
-		 */
-		if ( apply_filters( 'jetpack_gutenberg_cdn', false ) ) {
-			$cdn_base    = 'https://s0.wp.com/wp-content/mu-plugins/jetpack/_inc/blocks';
-			$view_script = "$cdn_base/view.js";
-			$view_style  = "$cdn_base/view$rtl.css";
-
-			/**
-			 * Filter to modify cache busting for Gutenberg block assets loaded from CDN
-			 *
-			 * @since 6.5.0
-			 *
-			 * @param string
-			 */
-			$version = apply_filters( 'jetpack_gutenberg_cdn_cache_buster', sprintf( '%s-%s', gmdate( 'd-m-Y' ), JETPACK__VERSION ) );
-		} else {
-			$view_script = plugins_url( '_inc/blocks/view.js', JETPACK__PLUGIN_FILE );
-			$view_style  = plugins_url( "_inc/blocks/view$rtl.css", JETPACK__PLUGIN_FILE );
-			$version     = Jetpack::is_development_version() && file_exists( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
-				? filemtime( JETPACK__PLUGIN_DIR . '_inc/blocks/view.js' )
-				: JETPACK__VERSION;
+	public static function load_assets_as_required( $type ) {
+		// Enqueue styles
+		$style_relative_path = '_inc/blocks/'. $type .'/view.'. ( is_rtl() ? '.rtl' : '' ) .'css';
```

Comment:
I think there is an issue with the `.` for the rtl file here. You're going to end up with a path like `_inc/blocks/tiled-gallery/view..rtlcss`